### PR TITLE
Return Scope object instead of unique_ptr<Scope> from Tracer::WithActiveSpan()

### LINKED ISSUE #787: why nostd::unique_ptr<Scope>  not Scope
Any reason why WithActiveSpan returns `nostd::unique_ptr<Scope>` instead of simply returning Scope?

### PR DESCRIPTION
**Pull request by lalitb**
Fixes #787 

## Changes

- As mentioned in #787 , `Tracer::WithActiveSpan()` can avoid memory allocation on heap by returning newly created `Scope` object instead of `unique_ptr<Scope>`. The copy constructor of `Scope` is implicitly deleted because of `unique_ptr` member variable, so this ensures multiple copies are not possible for `Scope`.

- Also modified etw::tracer::StartSpan() method to not make span active immediately after creating as this deviates from the [specs](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#span-creation). `tracer::WithActiveSpan()` should be called explicitly by instrumentation library to make it active. ( cc @maxgolov )

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

**Comment by lalitb** [comment]
@maxgolov - ETW exporter build is failing with this change due to the way we are using the `Scope` object in Tracer SDK there. Can you please review this PR whether it is a blocker for ETW?

**Review by pyohannes — APPROVED** [review]

File: exporters/etw/test/etw_tracer_test.cc [hunk 1]
```diff
@@ -116,13 +116,13 @@ TEST(ETWTracer, TracerCheck)
         };
         EXPECT_NO_THROW(innerSpan->AddEvent(eventName3, event3));
         std::this_thread::sleep_for (std::chrono::seconds(1));
-        innerSpan->End();
```
Comment (ThomsonTan) [inline exporters/etw/test/etw_tracer_test.cc]: Wondering why this is broken?

**Review by jsuereth — APPROVED** [review]
We may want to have a secondary method to apply active-span to context that DOESN'T have auto-scope (for advanced use cases).

Like this cleanup!

File: exporters/etw/test/etw_tracer_test.cc (hunk 1 above)
Comment (lalitb) [reply exporters/etw/test/etw_tracer_test.cc]: Looks like this comment is on an older / outdated commit?

Comment (ThomsonTan) [reply exporters/etw/test/etw_tracer_test.cc]: Yes. I was commenting on the line which added `EXPECT_NO_THROW`.

**Review by ThomsonTan — APPROVED** [review]

File: exporters/etw/test/etw_tracer_test.cc (hunk 1 above)
Comment (lalitb) [reply exporters/etw/test/etw_tracer_test.cc]: Ok got it. That was to keep with consistent with other End() calls,

**Pull request merged**